### DISCUSSION TITLE
Elementor Pro updated support for Pods 2.9+ -- Simple fix

### DISCUSSION THREAD

**Discussion by sc0ttkclark** [elementor/elementor#19359] [Feature Request]
This one is easy, I hope your developers can accommodate the simple one-line change to support a new feature in Pods 2.9+ that will be released in a few days.

I am the Lead Developer of the Pods Framework, we have some changes coming in Pods 2.9 that support more fine tuned image file types being selected for our File fields. There needs to be a change in an **Elementor Pro** file to support this properly.

The file is: `elementor-pro/modules/dynamic-tags/pods/module.php`

On line 99, this:

`if ( 'images' !== $field['options']['file_type'] ) {`

Should be changed to:

`if ( ! in_array( $field['options']['file_type'], [ 'images', 'images-any', 'any', 'other' ], true ) ) {`

We added support for an images-any option in Pods 2.9+ -- it works the same as images, but it allows for any "image/*" mime type supported by WordPress instead of just the four we had support for.

Please pass this along to your developers for a future **Elementor Pro** maintenance release
from me. It will be important as our shared users begin updating to Pods 2.9 this month.

**Comment by JoryHogeveen** [comment]
In addition to this I would also recommend supporting the `any` and `other` options since these could also contain image extensions.

**Reply by sc0ttkclark** [reply]
Updated my code to match that

**Comment by gursuj** [comment]
Found that a bug report for this (#29454) was closed due to security concerns. I assume allowing at least `images-any` should be safe, other than for svgs? Might be fine though since uploading files for Pods fileds is restricted to admins / trusted users?

Currently, image fields are unusable as dynamic tags in Elementor if their file type is set to `images-any` in Pods. This bug hasn't been documented in the [Pods docs](https://docs.pods.io/fields/file/) or the [Elementor Pods Integration docs](https://elementor.com/help/elementor-integration-with-pods/). There's already been a few online posts about issues likely caused by this ([1](https://www.reddit.com/r/elementor/comments/bc9x5h/pods_image_field_problems/), [2](https://www.reddit.com/r/elementor/comments/1dbuh3c/custom_taxonomy_fields_not_showing_in_select/), [3](https://wordpress.org/support/topic/pods-gallery-field-not-showing-up-in-elementor-gallery-widget/))   

@sc0ttkclark Moving this to 'Feature Requests' instead of 'Developer Community' in the top-right might help get more traction on this?

**Reply by sc0ttkclark** [reply]
@dinesh-58 moved it to Feature Requests just in case.

Might be worth checking with @nicholaszein to get clarification on why allowing any image mimetype that WordPress supports is a security issue. We use `get_allowed_mime_types()` and allow any `image/*` within that allow-list.

Right now that list which WordPress works from by default includes these:

```php
			// Image formats.
			'jpg|jpeg|jpe'                 => 'image/jpeg',
			'gif'                          => 'image/gif',
			'png'                          => 'image/png',
			'bmp'                          => 'image/bmp',
			'tiff|tif'                     => 'image/tiff',
			'webp'                         => 'image/webp',
			'avif'                         => 'image/avif',
			'ico'                          => 'image/x-icon',

			// TODO: Needs improvement. All images with the following mime types seem to have .heic file extension.
			'heic'                         => 'image/heic',
			'heif'                         => 'image/heif',
			'heics'                        => 'image/heic-sequence',
			'heifs'                        => 'image/heif-sequence',
```
